components/dialog/dialog-fullscreen.js

```diff
@@ -25,68 +27,151 @@ class DialogFullscreen extends LocalizeCoreElement(AsyncContainerMixin(DialogMix
+				.d2l-dialog-content {
+					display: grid;
+				}
 
-			.d2l-dialog-content-loading {
-				text-align: center;
-			}
+				.d2l-dialog-content > div {
+					/* required to properly calculate preferred height when there are bottom
+					margins at the end of the slotted content */
+					border-bottom: 1px solid transparent;
+					box-sizing: border-box;
+					height: 100%;
+					padding-bottom: 1rem;
+					padding-top: 1rem;
+				}
 
-			dialog.d2l-dialog-outer,
-			div.d2l-dialog-outer {
-				border: none;
-				border-radius: 0;
-				box-shadow: none;
-				height: 100%;
-				max-height: initial; /* required to override Chrome native positioning */
-				max-width: initial; /* required to override Chrome native positioning */
-				top: 0;
-				width: 100%;
-			}
+				.d2l-dialog-header > div > d2l-button-icon {
+					flex: none;
+					margin: -4px -15px 0 15px;
+				}
 
-			.d2l-dialog-footer.d2l-footer-no-content {
-				display: none;
+				:host([dir="rtl"]) .d2l-dialog-header > div > d2l-button-icon {
+					margin-left: -15px;
+					margin-right: 15px;
+				}
+				
+				dialog.d2l-dialog-outer,
+				div.d2l-dialog-outer {
+					border-radius: 8px;
+					height: calc(100% - 3rem);
+					margin: 1.5rem;
+					max-width: 1170px;
+					opacity: 0;
+					top: 0;
+					transform: translateY(-50px) scale(0.97);
+					transition: transform 200ms ease-out, opacity 200ms ease-out;
+					width: auto;
+				}
+
+				/* for screens wider than 1170px + 60px margins */
+				@media (min-width: 1230px) {
+					dialog.d2l-dialog-outer,
+					div.d2l-dialog-outer {
+						/* center the dialog */
+						margin-left: auto;
+						margin-right: auto;
+					}
+				}
+
+				:host([_state="showing"]) dialog.d2l-dialog-outer,
+				:host([_state="showing"]) div.d2l-dialog-outer {
+					opacity: 1;
+					transition-duration: 400ms;
+				}
+
+				dialog::backdrop {
+					transition: opacity 200ms ease-out;
+				}
+
+				:host([_state="showing"]) dialog::backdrop {
+					transition-duration: 400ms;
+				}
+
+				.d2l-dialog-footer {
+					border-top: 1px solid var(--d2l-color-gypsum);
+					padding-bottom: 0; /* 0.9rem padding included on button */
+					padding-top: 0.9rem;
+				}
+
+				@media (prefers-reduced-motion: reduce) {
+
+					dialog.d2l-dialog-outer,
+					div.d2l-dialog-outer {
+						transition: none;
+					}
+
+					dialog::backdrop {
+						transition: none;
+					}
+				}
 			}
 
 			@media (max-width: 615px) {
 
+				.d2l-dialog-header {
+					padding-bottom: 15px;
+				}
+
 				.d2l-dialog-header > div > d2l-button-icon {
+					flex: none;
 					margin: -8px -13px 0 15px;
 				}
 
+				.d2l-dialog-footer.d2l-footer-no-content {
+					padding: 0 0 5px 0;
+				}
+
+				.d2l-dialog-content > div {
+					/* required to properly calculate preferred height when there are bottom
+					margins at the end of the slotted content */
+					border-bottom: 1px solid transparent;
+					/* required to render full height in an i-Frame */
+					height: calc(100% - 1px);
```

Comment:
We need the -1px so that iFrames can render at the full height, but only display the scrollbar when there is an overflow (100% will always display a scrollbar, even with short contents)